provider/datagen/src/transform/cldr/units/helpers.rs

```diff
@@ -0,0 +1,363 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::str::FromStr;
+
+use icu_provider::DataError;
+use icu_unitsconversion::provider::{ConstantExactness, Sign};
+use num_bigint::BigInt;
+use num_rational::BigRational;
+
+/// Converts a decimal number represented as a string into a BigRational.
+/// Examples:
+/// - "1" is converted to 1/1
+/// - "1.5" is converted to 15/10
+/// - "1.05" is converted to 105/100
+/// - "1.005" is converted to 1005/1000
+/// - "1.5.5" is an invalid decimal number
+/// NOTE:
+/// - "1." is not a valid decimal number
+/// - BigRational represents a rational number in the simplest form. For example, 15/10 is converted to 3/2.
+pub fn convert_decimal_to_bigrational(decimal: &str) -> Result<BigRational, DataError> {
+    let parts: Vec<&str> = decimal.split('.').collect();
+    match parts.len() {
+        1 => BigRational::from_str(parts[0])
+            .map_err(|_| DataError::custom("the integer-part is not a valid number")),
+        2 => {
+            let numerator = BigInt::from_str(parts.join("").as_str()).map_err(|_| {
+                DataError::custom("the integer-part and fractional-part are not a valid number")
+            })?;
+            let denominator = BigInt::from(10u32).pow(parts[1].len() as u32);
+            Ok(BigRational::new(numerator, denominator))
+        }
+        _ => Err(DataError::custom("the base is not a valid number")),
+    }
```

Comment:
nit: this can be done without an allocation or panicking indexing. We don't require this in datagen, but I want to mention it

```suggestion
    let parts = decimal.split('.');
    match (parts.next(), parts.next(), parts.next()) {
        (Some(int), None, None) => BigRational::from_str(int)
            .map_err(|_| DataError::custom("the integer-part is not a valid number")),
        (Some(int), Some(fract), None) => {
            let numerator = BigInt::from_str(format!("{int}{fract}").map_err(|_| {
                DataError::custom("the integer-part and fractional-part are not a valid number")
            })?;
            let denominator = BigInt::from(10u32).pow(fract.len() as u32);
            Ok(BigRational::new(numerator, denominator))
        }
        _ => Err(DataError::custom("the base is not a valid number")),
    }
```

Reply:
There is no `next` in `Vec<&str>`, but I used `first` with `unwrap`.

Using `switch` really makes things obvious.

Reply:
If you don't collect into the `Vec` there's the `next` on the iterator.